src/sample-executors/RelayOrderExecutor.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {Owned} from "solmate/src/auth/Owned.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {SignedOrder} from "UniswapX/src/base/ReactorStructs.sol";
+import {CurrencyLibrary} from "UniswapX/src/lib/CurrencyLibrary.sol";
+import {Permit2Lib} from "permit2/src/libraries/Permit2Lib.sol";
+import {Multicall} from "./Multicall.sol";
+import {IRelayOrderReactor} from "../interfaces/IRelayOrderReactor.sol";
+
+/// @notice Sample executor for Relay orders
+contract RelayOrderExecutor is Multicall, Owned {
+    using SafeTransferLib for ERC20;
+    using CurrencyLibrary for address;
+
+    /// @notice thrown if this contract is called by an address other than the whitelisted caller
+    error CallerNotWhitelisted();
+
+    address private immutable whitelistedCaller;
+    IRelayOrderReactor private immutable reactor;
+
+    modifier onlyWhitelistedCaller() {
+        if (msg.sender != whitelistedCaller) {
+            revert CallerNotWhitelisted();
+        }
+        _;
+    }
+
+    constructor(address _whitelistedCaller, IRelayOrderReactor _reactor, address _owner) Owned(_owner) {
+        whitelistedCaller = _whitelistedCaller;
```

Comment:
We should probably allow changing the whitelistedCaller? Or allow execute/executBatch to be callable by the owner as well

Reply:
Yea so in the executor we use in prod for UniswapX today we actually allow for any of 10 whitelisted callers. That was mostly a requirement for scaling - but can consider a better design here.

Reply:
Let's just allow for owner to change caller in this version and can change if needed